Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServiceTest.cs

```diff
@@ -407,11 +432,111 @@ public async Task DownloadCache_RequestUrl(string hostUrl, string downloadUrl)
+        {
+            var downloaderMock = Mock.Of<IDownloader>(x => x.DownloadStream(It.IsAny<Uri>()) == Task.FromException<Stream>(new HttpRequestException()));
+            sut = new SonarQubeWebService(downloaderMock, serverUrl, version, logger);
+
+            var localSettings = GetLocalSettings(ProjectKey, ProjectBranch);
+            var result = await sut.DownloadCache(localSettings);
+
+            result.Should().BeEmpty();
+            logger.AssertSingleWarningExists("Incremental PR analysis: an error occurred while deserializing the cache entries! Exception of type 'System.Net.Http.HttpRequestException' was thrown.");
+        }
+
+        [TestMethod]
+        public async Task DownloadCache_WhenCacheStreamReadThrows_ReturnsEmptyAndLogsException()
+        {
+            var streamMock = new Mock<Stream>();
+            streamMock.Setup(x => x.Length).Throws<InvalidOperationException>();
+            sut = new SonarQubeWebService(MockIDownloader(streamMock.Object), serverUrl, version, logger);
+            var localSettings = GetLocalSettings(ProjectKey, ProjectBranch);
+
+            var result = await sut.DownloadCache(localSettings);
+
+            result.Should().BeEmpty();
+            logger.AssertSingleWarningExists("Incremental PR analysis: an error occurred while deserializing the cache entries! Operation is not valid due to the current state of the object.");
+        }
+
+        [TestMethod]
+        public async Task DownloadCache_WhenCacheStreamDeserializeThrows_ReturnsEmptyAndLogsException()
+        {
+            var invalidProtoStream = new MemoryStream(new byte[] { 42, 42 }); // this is a random byte array that fails deserialization
+            sut = new SonarQubeWebService(MockIDownloader(invalidProtoStream), serverUrl, version, logger);
+            var localSettings = GetLocalSettings(ProjectKey, ProjectBranch);
+
+            var result = await sut.DownloadCache(localSettings);
+
+            result.Should().BeEmpty();
+            logger.AssertSingleWarningExists("Incremental PR analysis: an error occurred while deserializing the cache entries! While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.");
+        }
+
+        private static Stream CreateCacheStream(IMessage message)
+        {
+            var stream = new MemoryStream();
+            message.WriteDelimitedTo(stream);
+            stream.Seek(0, SeekOrigin.Begin);
+            return stream;
+        }
+
+        private static IDownloader MockIDownloader(Stream stream) =>
+            Mock.Of<IDownloader>(x => x.DownloadStream(It.IsAny<Uri>()) == Task.FromResult(stream));
+
+        private ProcessedArgs GetLocalSettings(string projectKey, string branch, string organization = "placeholder", string token = "placeholder")
```

Comment:
```suggestion
        private static ProcessedArgs CreateLocalSettings(string projectKey, string branch, string organization = "placeholder", string token = "placeholder")
```

Reply:
I think you did this for Cloud test but missed the occurrence from SonarQube tests.